src/app/widgets/exchange/exchange.component.ts

```diff
@@ -1,10 +1,25 @@
-import { Component } from '@angular/core';
+import { CurrenciesService } from "./services/currenciesList.service";
+import { Exchange } from "./interfaces/Currency";
+import { Component } from "@angular/core";
+import { Router } from "@angular/router";
 
 @Component({
-  selector: 'app-exchange',
-  templateUrl: './exchange.component.html',
-  styleUrls: ['./exchange.component.scss']
+  selector: "app-exchange",
+  templateUrl: "./exchange.component.html",
+  styleUrls: ["./exchange.component.scss"],
+  providers: [CurrenciesService],
 })
 export class ExchangeComponent {
+  selectedCurrency!: number;
 
+  exchange!: Exchange[];
+
+  constructor(private router: Router) {}
+
+  navigateToCurrency() {
+    this.router.navigate([""]);
+  }
+  navigateToCurrency2() {
```

Comment:
Avoid using numbers in function/method names as well for any naming. It's not readable ant not easy to maintain